beacon-chain/rpc/helpers/broadcast.go

```diff
@@ -0,0 +1,173 @@
+	for _, opt := range options {
+		opt(opts)
+	}
+
+	verifiedRODataColumns := make([]blocks.VerifiedRODataColumn, 0, len(sidecars))
+
+	// Create verified RO data columns
+	for _, sd := range sidecars {
+		roDataColumn, err := blocks.NewRODataColumnWithRoot(sd, root)
+		if err != nil {
+			return errors.Wrap(err, "new read-only data column with root")
+		}
+
+		// We build this block ourselves, so we can upgrade the read only data column sidecar into a verified one.
+		verifiedRODataColumn := blocks.NewVerifiedRODataColumn(roDataColumn)
+		verifiedRODataColumns = append(verifiedRODataColumns, verifiedRODataColumn)
+	}
+
+	// Broadcast data columns concurrently
+	eg, _ := errgroup.WithContext(ctx)
+	for _, sd := range sidecars {
+		// Copy the iteration instance to a local variable to give each go-routine its own copy to play with.
+		// See https://golang.org/doc/faq#closures_and_goroutines for more details.
+		sidecar := sd
+		eg.Go(func() error {
+			// Compute the subnet index based on the column index.
+			subnet := peerdas.ComputeSubnetForDataColumnSidecar(sidecar.Index)
+
+			if err := broadcastFunc(root, subnet, sidecar); err != nil {
+				return errors.Wrap(err, "broadcast data column")
+			}
+
+			return nil
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return errors.Wrap(err, "wait for data columns to be broadcasted")
+	}
+
+	// Optionally receive the data columns
+	if opts.onReceiveDataColumns != nil {
+		if err := opts.onReceiveDataColumns(verifiedRODataColumns); err != nil {
+			return errors.Wrap(err, "receive data column")
+		}
+	}
+
+	// Call the callback if provided
+	if opts.onDataColumnsProcessed != nil {
+		opts.onDataColumnsProcessed(verifiedRODataColumns)
+	}
+
+	return nil
+}
+
+// BlobOption configures options for broadcasting blob sidecars.
+type BlobOption func(*blobOptions)
+
+type blobOptions struct {
+	onReceiveBlob   func(ctx context.Context, blob blocks.VerifiedROBlob) error
+	onBlobProcessed func(blob blocks.VerifiedROBlob)
+}
+
+// WithBlobReceiver sets the callback for receiving blobs.
+func WithBlobReceiver(fn func(ctx context.Context, blob blocks.VerifiedROBlob) error) BlobOption {
+	return func(o *blobOptions) {
+		o.onReceiveBlob = fn
+	}
+}
+
+// WithBlobProcessedCallback sets the callback to be called after a blob is processed.
+func WithBlobProcessedCallback(fn func(blob blocks.VerifiedROBlob)) BlobOption {
+	return func(o *blobOptions) {
+		o.onBlobProcessed = fn
+	}
+}
+
+// BroadcastBlobSidecars broadcasts blob sidecars concurrently and optionally receives them.
+func BroadcastBlobSidecars(
+	ctx context.Context,
+	sidecars []*ethpb.BlobSidecar,
+	root [32]byte,
+	broadcastFunc BlobBroadcastFunc,
+	options ...BlobOption,
+) error {
+	opts := &blobOptions{}
+	for _, opt := range options {
+		opt(opts)
+	}
+
+	eg, eCtx := errgroup.WithContext(ctx)
+	for i, sc := range sidecars {
+		// Copy the iteration instance to a local variable to give each go-routine its own copy to play with.
```

Comment:
Is this precaution still useful since go 1.22? https://go.dev/blog/loopvar-preview